src/cdk/v2/destinations/reddit/utils.test.js

```diff
@@ -0,0 +1,121 @@
+const { calculateRevenue, calculateRevenueField } = require('./utils');
+
+describe('calculateRevenue', () => {
+  // Calculates revenue for a single product with defined price and quantity
+  it('should calculate revenue for a single product with defined price and quantity', () => {
+    const properties = {
+      price: 10,
+      quantity: 2,
+    };
+
+    const result = calculateRevenue(properties);
+
+    expect(result).toBe(20);
+  });
+
+  // Returns null for properties parameter being undefined
+  it('should return null for price parameter being undefined', () => {
+    const properties = { products: [{ quantity: 1 }] };
+
+    const result = calculateRevenue(properties);
+
+    expect(result).toBeNull();
+  });
+
+  // Calculates revenue for a single product with defined price and default quantity
+  it('should calculate revenue for a single product with defined price and default quantity', () => {
+    const properties = {
+      price: 10,
+    };
+
+    const result = calculateRevenue(properties);
+
+    expect(result).toBe(10);
+  });
+
+  // Calculates revenue for multiple products with defined prices and quantities
+  it('should calculate revenue for multiple products with defined prices and quantities', () => {
+    const properties = {
+      products: [{ price: 10, quantity: 2 }, { quantity: 3 }],
+    };
+
+    const result = calculateRevenue(properties);
+
+    expect(result).toBe(20);
```

Comment:
shouldn't it be 2000?

Reply:
umm no, because it is calculated as (10 x 2) + (3 x 0 ) = 20